### PR TITLE
COOK-4672 - Update library so it can be used in other libraries

### PR DESCRIPTION
**Pull request by spion06**

### CODE REVIEW

**Comment by martinisoft** [comment]
Can we see about getting this merged @jtimberman and @spion06 ?

Running into this trying to do some library refactoring.

File: libraries/chef_vault_item.rb [hunk 1]
```diff
@@ -20,28 +20,29 @@
...
-      begin
-        ChefVault::Item.load(bag, item)
-      rescue ChefVault::Exceptions::KeysNotFound
-        Chef::DataBagItem.load(bag, item)
-      end
-    end
-  end
+   begin
+     ChefVault::Item.load(bag, item)
+   rescue ChefVault::Exceptions::KeysNotFound
```
Comment (martinisoft) [inline libraries/chef_vault_item.rb]: Think this line changes the current master version rescue which is:

```ruby
rescue ChefVault::Exceptions::KeysNotFound, ChefVault::Exceptions::SecretDecryption
```

**Comment by jtimberman** [comment]
I rebased this and it's on master, 04747260ff4ea0ccb769ba4c5b0bda784e331899

**Comment by spion06** [comment]
thanks @jtimberman and @martinisoft. I forgot about this pull

**Comment by jtimberman** [comment]
No problem. Thanks for contributing as I was looking to do this myself the other day and then remembered this PR :smile_cat: @spion06

**Pull request closed without merging**